packages/x-charts-premium/src/ChartsRenderer/ChartsRenderer.tsx

```diff
@@ -0,0 +1,308 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { BarChart } from '@mui/x-charts/BarChart';
+import { LineChart } from '@mui/x-charts/LineChart';
+import { PieChart } from '@mui/x-charts/PieChart';
+import { configurationOptions } from './configuration';
+import { colorPaletteLookup } from './colors';
+
+const getLegendPosition = (position: string) => {
+  switch (position) {
+    case 'top':
+      return { vertical: 'top' as const };
+    case 'bottom':
+      return { vertical: 'bottom' as const };
+    case 'left':
+      return { horizontal: 'start' as const };
+    case 'right':
+      return { horizontal: 'end' as const };
+    default:
+      return undefined;
+  }
+};
+
+export interface ChartsRendererProps {
+  categories: { id: string; label: string; data: (string | number | null)[] }[];
+  series: { id: string; label: string; data: (number | null)[] }[];
+  chartType: string;
+  configuration: Record<string, any>;
+  onRender?: (
+    type: string,
+    props: Record<string, any>,
+    Component: React.ComponentType<any>,
+  ) => React.ReactNode;
+}
+
+function ChartsRenderer({
```

Comment:
This component begs for JSDoc and a link to the integration docs, since there's nothing in its name that suggests its purpose.
Would be helpful for users discovering the `@mui/x-charts-premium/ChartsRenderer` export accidentally